packages/dashboards/src/lib/configurator/components/widgets/table/columns-editor-v2/column-configuration/table-column-configuration/table-column-configuration.component.ts

```diff
@@ -0,0 +1,92 @@
+import { ChangeDetectorRef, Component, forwardRef, Input, OnInit } from "@angular/core";
+import { AbstractControl, ControlValueAccessor, FormBuilder, FormControl, FormGroup, NG_VALIDATORS, NG_VALUE_ACCESSOR, ValidationErrors } from "@angular/forms";
+import { Subject } from "rxjs";
+import { takeUntil } from "rxjs/operators";
+
+import { ITableWidgetColumnConfig } from "../../../../../../../components/table-widget/types";
+
+@Component({
+    selector: "nui-table-column-configuration",
+    templateUrl: "./table-column-configuration.component.html",
+    providers: [
+        {
+            provide: NG_VALUE_ACCESSOR,
+            useExisting: forwardRef(() => TableColumnConfigurationComponent),
+            multi: true,
+        },
+        {
+            provide: NG_VALIDATORS,
+            useExisting: forwardRef(() => TableColumnConfigurationComponent),
+            multi: true,
+        },
+    ],
+})
+export class TableColumnConfigurationComponent implements ControlValueAccessor, OnInit {
+    public static lateLoadKey = "TableColumnConfigurationComponent";
+
+    public form: FormGroup;
+    public changeFn: Function;
+
+    private destroy$ = new Subject();
+    private input: ITableWidgetColumnConfig;
+
+    @Input() formControl: AbstractControl;
+    @Input() isWidthMessageDisplayed = false;
+
+    constructor(private formBuilder: FormBuilder, public changeDetector: ChangeDetectorRef) {
+        this.form = this.formBuilder.group({
+            description: this.formBuilder.control({}),
+            presentation: this.formBuilder.control({}),
+        });
+
+        this.form.valueChanges.pipe(takeUntil(this.destroy$)).subscribe((value) => {
+            const newValue = {
+                id: this.input?.id,
+                ...value.description,
+                formatter: value.presentation,
+            };
+            this.changeFn(newValue);
+        });
+    }
+
+    public ngOnInit(): void {
+        const origFunc = this.formControl.markAsTouched;
```

Comment:
It would be good to add a comment at all the spots where you are hijacking markAsTouched with the reason for doing it.